### PR TITLE
add in all missing descriptions

### PR DESCRIPTION
**Pull request by nicolaskruchten**
If we merge in https://github.com/plotly/dash-core-components/pull/542 then this PR resolves all "missing description" warnings. If we don't merge that one we need to write a description for this mysterious `key` :)

See also https://github.com/plotly/dash-core/issues/52

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
Note that I wrote very few of these: they're mostly copy-pasta from other places where similar props were used :)

File: src/components/DatePickerRange.react.js [hunk 1]
```diff
@@ -187,7 +192,14 @@ DatePickerRange.propTypes = {
...
     start_date: PropTypes.string,
 
+    /**
+     * The ID of the start date input field
+     */
     start_date_id: PropTypes.string,
+
+    /**
+     * The ID of the end date input field
+     */
```
Comment (alexcjohnson) [inline src/components/DatePickerRange.react.js]: I might elaborate a bit, something like:

> The HTML element ID of the end date input field. Not used by dash, only by CSS.

File: src/components/Graph.react.js [hunk 2]
```diff
@@ -75,6 +75,12 @@ function generateId() {
     );
 }
 
+/**
+ * Graph cand be used to render any plotly.js-powered data visualization.
```
Comment (alexcjohnson) [inline src/components/Graph.react.js]: can~d~

File: src/components/Input.react.js [hunk 3]
```diff
@@ -178,6 +178,9 @@ Input.propTypes = {
         PropTypes.bool,
     ]),
 
+    /**
+     * Provides a hint as to the type of data that might be entered by the user while editing the element or its contents.
```
Comment (alexcjohnson) [inline src/components/Input.react.js]: so it's clear why:

> Provides a hint *to the browser* as to the type...

File: src/components/Link.react.js [hunk 4]
```diff
@@ -27,6 +27,9 @@ function CustomEvent(event, params) {
 }
 CustomEvent.prototype = window.Event.prototype;
 
+/**
+ * Link allows you to create a clickable link within a multi-page app
+ */
```
Comment (alexcjohnson) [inline src/components/Link.react.js]: Do we want to say "for links outside this app, use `html.A` instead"? Or is `Link` good in both cases?

**Comment by nicolaskruchten** [comment]
@alexcjohnson addressed all your comments.

**Pull request merged**